src/Player/Player.cpp

```diff
@@ -34,6 +35,111 @@ namespace Player {
         Terminal::debug("Player fortify");
+        Terminal::print("Fortification phase");
+        vector<string> selections;
+        bool validChoice = false;
+        int source_country_index;
+        Country * source_country;
+
+        bool can_fortify = false;
+        // confirm fortification is possible
+        for (auto & country : *countries) {
+            for (auto & neighbor : *country->get_neighbors()) {
+                if (this == neighbor->get_owner() && country->get_armies() > 1) {
+                    can_fortify = true;
+                    break;
+                }
+            }
+            if (can_fortify) {
+                break;
+            }
+        }
+
+        if (!can_fortify) {
+            Terminal::print("Fortification is not possible, you do not own two valid neighboring countries");
+            return;
+        } else {
+            Terminal::print("Fortification is possible, would you like to fortify?");
+            vector<string> fortify;
+            fortify.emplace_back("Yes");
+            fortify.emplace_back("No");
+            int response = Terminal::print_select(fortify);
+            if (response == 1) {
+                return;
+            }
+        }
+
+
+        Terminal::print("Select source country");
+        for (auto & country : *countries) {
+            selections.push_back(country->to_string_with_neighbors());
+        }
+        while (!validChoice) {
+            source_country_index = Terminal::print_select(selections);
+            source_country = countries->at(source_country_index);
+            if (source_country->get_armies() > 1) {
+                for (auto & country : *source_country->get_neighbors()) {
+                    if (this == country->get_owner()) {
+                        validChoice = true;
+                        break;
+                    }
+                }
+                if (!validChoice) {
+                    Terminal::print("must choose a country with at least one neighbor which you own.");
+                }
+            } else {
+                Terminal::print("must choose a country which has at least 2 armies.");
+            }
+        }
+
+        selections.clear();
+        selections.shrink_to_fit();
+        vector<Country *> * source_country_neighbors = source_country->get_neighbors();
+        int target_country_index;
+        Country * target_country;
+        validChoice = false;
+        Terminal::print("Select target country");
+        for (Country * neighbor: *source_country_neighbors) {
+            selections.push_back(neighbor->to_string_with_neighbors());
+        }
+        while (!validChoice) {
+            target_country_index = Terminal::print_select(selections);
+            target_country = source_country_neighbors->at(target_country_index);
+            if (this != target_country->get_owner()) {
+                validChoice = false;
+                Terminal::print("Must own selected country");
```

Comment:
You should show only user's country as possible target country

Reply:
I debated this, as I user, I might want to know who all of my neighbors are so I can make an informed decision (Like will moving too many armies result in a country that is poorly defended)

Reply:
Make sense. I though you had a special to string to show neighbor.